go/genesis/genesis_test.go

```diff
@@ -412,6 +436,53 @@ func TestGenesisSanityCheck(t *testing.T) {
 	d.Registry.Entities = []*entity.SignedEntity{signedTestEntity}
 	require.NoError(d.SanityCheck(), "test entity should pass")
 
+	// NOTE: We will break the staking part of testDoc and gradually fix it to
+	// test all related registry & staking genesis checks.
+	d = *testDoc
+	d.Staking.Ledger = make(map[signature.PublicKey]*staking.Account)
+	d.Staking.Delegations = make(map[signature.PublicKey]map[signature.PublicKey]*staking.Delegation)
+
+	d.Registry.Entities = []*entity.SignedEntity{signedTestEntity}
+	err = d.SanityCheck()
+	require.NotNil(err, "entity without an account should be rejected")
+	require.Regexp("no account associated with entity with ID", err)
```

Comment:
I think this will be the first we have of this kind of check

Reply:
We already have 2 checks like that :slightly_smiling_face: :
https://github.com/oasislabs/oasis-core/blob/798fbe38f4665a1cb78cec2d605bf84dcfa51f95/go/worker/common/host/sandboxed_test.go#L222-L234

Reply:
From my experience, just checking for a presence of an error and not checking for occurrence of a particular error is quite bad.

Frequently, a developer will break code in a new way and the tests will seem to pass even if they shouldn't because the error that occurred is not the one that was expected.

I plan to make a follow-up PR to update these tests to actually check for particular errors (and not just any error).